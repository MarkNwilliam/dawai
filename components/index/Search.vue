<template>
  <v-row align="center" justify="center">
    <v-col cols="12" xs="12">
      <h1 class="text-center">* This is a demo</h1>
    </v-col>
    <v-col cols="12" xs="12" md="8" lg="5">
      <v-autocomplete
        v-model="select"
        color="primary"
        label="Search for Medicine"
        hint="Try 'Panadol'..."
        :items="items"
        :loading="loading"
        :search-input.sync="search"
        :prepend-inner-icon="mdiPill"
        :clear-icon="mdiClose"
        loader-height="3"
        hide-details
        hide-no-data
        cache-items
        clearable
        autofocus
        solo
      ></v-autocomplete>
    </v-col>
    <v-col class="text-center" cols="12" xs="12" md="12" lg="1">
      <v-btn
        color="primary"
        :loading="isLoading"
        :disabled="!select"
        large
        rounded
        @click="searchMedicine"
        ><v-icon>{{ mdiMagnify }}</v-icon></v-btn
      >
    </v-col>
    <v-col v-if="isLoading" class="mt-12" cols="12" xs="12">
      <v-sheet color="transparent" class="px-3 pt-3 pb-3">
        <v-skeleton-loader
          class="slight-br my-3"
          max-width="auto"
          type="list-item-avatar-three-line"
        ></v-skeleton-loader>
        <v-skeleton-loader
          class="slight-br my-3"
          max-width="auto"
          type="list-item-avatar-three-line"
        ></v-skeleton-loader>
        <v-skeleton-loader
          class="slight-br my-3"
          max-width="auto"
          type="list-item-avatar-three-line"
        ></v-skeleton-loader>
        <v-skeleton-loader
          class="slight-br"
          max-width="auto"
          type="image"
        ></v-skeleton-loader>
      </v-sheet>
    </v-col>
  </v-row>
</template>

<script>
import {
  mdiPill,
  mdiMagnify,
  mdiClose,
  mdiChevronDown,
  mdiBullseyeArrow,
  mdiFormatListBulletedType
} from '@mdi/js'
export default {
  inject: ['theme'],
  data() {
    return {
      select: null,
      search: '',
      loading: false,
      isLoading: false,
      items: [],
      mdiPill,
      mdiMagnify,
      mdiClose,
      mdiChevronDown,
      mdiBullseyeArrow,
      mdiFormatListBulletedType,
      medicine: [
        'Panadol',
        'Panadol Extra',
        'Panadol Light',
        'Ibuprofen',
        'Ibuprofen Extra',
        'Paracetamol',
        'Super Paracetamol'
      ]
    }
  },
  watch: {
    search(val) {
      val && val !== this.select && this.querySelections(val)
    }
  },
  methods: {
    searchMedicine() {
      this.isLoading = true
      setTimeout(() => {
        this.isLoading = false
      }, 5000)
    },
    querySelections(v) {
      this.loading = true
      setTimeout(() => {
        this.items = this.medicine.filter((e) => {
          return (e || '').toLowerCase().includes((v || '').toLowerCase())
        })
        this.loading = false
      }, 1000)
    }
  }
}
</script>
