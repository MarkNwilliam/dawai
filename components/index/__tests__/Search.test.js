// Libraries
import Vuetify from 'vuetify'

// Component
import { mount, createLocalVue } from '@vue/test-utils'
import Search from '../Search'

// Utilities
const localVue = createLocalVue()

describe('Search.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should have a custom search and match snapshot', done => {
    const wrapper = mount(Search, {
      localVue,
      vuetify
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    const alert = wrapper.find('v-alert')
    console.log(alert)

    expect(alert.text()).toBe('This is a demo')
    done()
  })
})
