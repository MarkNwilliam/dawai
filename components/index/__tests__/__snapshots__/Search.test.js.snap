// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue should have a custom search and match snapshot 1`] = `
"<v-col cols=\\"12\\" sm=\\"12\\"><v-row justify=\\"center\\" align=\\"center\\"><v-col cols=\\"12\\" sm=\\"6\\"><v-alert icon=\\"M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z\\" border=\\"left\\" value=\\"true\\" type=\\"info\\" text=\\"\\">
        This is a demo
      </v-alert></v-col></v-row> <v-row align=\\"center\\" justify=\\"center\\"><v-col cols=\\"12\\" xs=\\"12\\" md=\\"8\\" lg=\\"5\\"><v-autocomplete color=\\"primary\\" label=\\"Search for Medicine\\" hint=\\"Try 'Panadol'...\\" items=\\"\\" search-input=\\"\\" prepend-inner-icon=\\"M4.22,11.29L11.29,4.22C13.64,1.88 17.43,1.88 19.78,4.22C22.12,6.56 22.12,10.36 19.78,12.71L12.71,19.78C10.36,22.12 6.56,22.12 4.22,19.78C1.88,17.43 1.88,13.64 4.22,11.29M5.64,12.71C4.59,13.75 4.24,15.24 4.6,16.57L10.59,10.59L14.83,14.83L18.36,11.29C19.93,9.73 19.93,7.2 18.36,5.64C16.8,4.07 14.27,4.07 12.71,5.64L5.64,12.71Z\\" clear-icon=\\"M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z\\" loader-height=\\"3\\" hide-details=\\"\\" hide-no-data=\\"\\" cache-items=\\"\\" clearable=\\"\\" autofocus=\\"\\" solo=\\"\\"></v-autocomplete></v-col> <v-col cols=\\"12\\" xs=\\"12\\" md=\\"12\\" lg=\\"1\\" class=\\"text-center\\"><v-btn color=\\"primary\\" disabled=\\"true\\" large=\\"\\" rounded=\\"\\"><v-icon>M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z</v-icon></v-btn></v-col> <!----> <!----></v-row></v-col>"
`;
